### PR TITLE
Add support for channel notation

### PR DESCRIPTION
**Pull request by Areedtz**
- [x] Add parsing of channel notation
- [x] Add handling of channel options in general types
- [x] Parse channel notation to channels in the main proverif process
- [x] Pass channels to the party processes in proverif
- [ ] Make parallel processes exactly like Alessandro showed on the whiteboard

### CODE REVIEW

File: src/types.ml [hunk 1]
```diff
@@ -50,23 +50,26 @@ let rec binds = function
...
   | DefGlobal of ident * (ident * principal) list * global_type * global_type
   | CallGlobal of ident * term list
   | GlobalEnd
 
 (* Local Type *)
 type local_type =
-    LSend of principal * term * local_type
-  | LRecv of principal * pattern * term * local_type
+    LSend of ident * channel_option * term * local_type
+  | LRecv of ident * channel_option * pattern * term * local_type
```
Comment (elit0451) [inline src/types.ml:72]: Why do we have `ident` here?

File: src/types.ml [hunk 2]
```diff
@@ -237,8 +246,8 @@ let rec print_sep = function
 let mscgen (pr:problem): unit =
   let last l = List.nth l (List.length l - 1) in
   let rec mscglobal = function
-  | Send(p, q, x, t, g) -> p ^ "->" ^ q ^ " [label=\"" ^ x ^ " = " ^ show_term t ^ "\"];\n" ^ mscglobal g
-  | Branch(p, q, t, branches) ->
+  | Send(p, q, _, x, t, g) -> p ^ "->" ^ q ^ " [label=\"" ^ x ^ " = " ^ show_term t ^ "\"];\n" ^ mscglobal g
```
Comment (elit0451) [inline src/types.ml:249]: Why do we just go with the public case in here? - `"->"`

File: src/types.ml (hunk 1 above)
Comment (Areedtz) [reply src/types.ml:72]: I have changed it to be a joined name of both the sender and the receiver.
This means that a message between Alice and Bob should end up having the ident "AliceBob". It should be sorted, meaning it always ends up being "AliceBob" even if Bob is the sender.

File: src/types.ml (hunk 2 above)
Comment (Areedtz) [reply src/types.ml:249]: Because I don't know if I will break the mscgen if I change it to something else ;)

**Pull request merged**